src/app/index.js

```diff
@@ -0,0 +1,131 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import classnames from 'classnames';
+import { connect } from 'react-redux';
+import { bindActionCreators } from 'redux';
+import Application from 'react-rainbow-components/components/Application';
+import Sidebar from 'react-rainbow-components/components/Sidebar';
+import SidebarItem from 'react-rainbow-components/components/SidebarItem';
+import Routes from '../routes';
+import SectionHeading from '../components/SectionHeading';
+import {
+    ApplicationIcon,
+    ChartsIcon,
+    DashboardIcon,
+    FormsIcon,
+    MessageIcon,
+    PagesIcon,
+    PuzzleIcon,
+} from '../components/icons';
+import history, { navigateTo } from '../history';
+import fetchSidebarState from '../redux/actions/app';
+import './styles.css';
+
+function resolveCurrentUrl() {
+    return history.location.pathname.split('/')[1] || 'dashboard';
+}
+
+class App extends Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            selectedItem: resolveCurrentUrl(),
+        };
+        this.handleOnSelect = this.handleOnSelect.bind(this);
+    }
+
+    getSidebarClassNames() {
+        const { isSidebarVisible } = this.props;
+        return classnames('react-rainbow-admin-app_sidebar', {
+            'react-rainbow-admin-app_sidebar--hidden': isSidebarVisible,
+        });
+    }
+
+    getRouterContainerClassNames() {
+        const { isSidebarVisible } = this.props;
+        return classnames('react-rainbow-admin-app_router-container', {
+            'react-rainbow-admin-app_router-container--sidebar-hidden': isSidebarVisible,
+        });
+    }
+
+    handleOnSelect(e, selectedItem) {
+        return this.setState({ selectedItem });
+    }
+
+    render() {
+        const { selectedItem } = this.state;
+        const { fetchSidebarState } = this.props;
+        return (
+            <Application>
+                <SectionHeading onToogleSidebar={fetchSidebarState} />
```

Comment:
the action must indicate the action
in this case you are not fetching sidebar state but toogleSidebar